tests/runtime/flavorful/host.ts

```diff
@@ -63,7 +65,15 @@ export async function run () {
   assert.deepStrictEqual(wasm.fListInVariant2(), "list_in_variant2");
   assert.deepStrictEqual(wasm.fListInVariant3("input3"), "output3");
 
-  assert.deepStrictEqual(wasm.errnoResult().tag, 'err');
+  try {
+    wasm.errnoResult();
+    assert.ok(false);
+  }
+  catch (e: any) {
```

Comment:
Do you know if there's a way in typescript to annotate the type of the error that can be thrown so `catch` doesn't need type annotations?

Reply:
Unfortunately TypeScript only permits `any | unknown` for errors, it simply doesn't handle error type flow at all.